crates/router/src/core/payouts.rs

```diff
@@ -0,0 +1,867 @@
+        state,
+        &merchant_account,
+        &payouts::PayoutRequest::PayoutRetrieveRequest(req.to_owned()),
+    )
+    .await?;
+
+    response_handler(
+        state,
+        &merchant_account,
+        &payouts::PayoutRequest::PayoutRetrieveRequest(req.to_owned()),
+        &payout_data,
+    )
+    .await
+}
+
+#[instrument(skip_all)]
+pub async fn payouts_cancel_core(
+    state: &AppState,
+    merchant_account: storage::merchant_account::MerchantAccount,
+    req: payouts::PayoutActionRequest,
+) -> RouterResponse<payouts::PayoutCreateResponse> {
+    let payout_data = make_payout_data(
+        state,
+        &merchant_account,
+        &payouts::PayoutRequest::PayoutActionRequest(req.to_owned()),
+    )
+    .await?;
+
+    let payout_create = payout_data.payout_create.to_owned();
+    let status = payout_create.status.foreign_into();
+
+    // Verify if fulfillment can be triggered
+    if helpers::is_payout_terminal_state(status) {
+        return Err(report!(errors::ApiErrorResponse::InvalidRequestData {
+            message: format!(
+                "{} is in terminal state [status: {}]",
+                payout_create.payout_id.to_owned(),
+                status
+            ),
+        }));
+    }
+
+    // TODO: Add connector integration
+    response_handler(
+        state,
+        &merchant_account,
+        &payouts::PayoutRequest::PayoutActionRequest(req.to_owned()),
+        &payout_data,
+    )
+    .await
+}
+
+#[instrument(skip_all)]
+pub async fn payouts_fulfill_core(
+    state: &AppState,
+    merchant_account: storage::merchant_account::MerchantAccount,
+    req: payouts::PayoutActionRequest,
+) -> RouterResponse<payouts::PayoutCreateResponse> {
+    let mut payout_data = make_payout_data(
+        state,
+        &merchant_account,
+        &payouts::PayoutRequest::PayoutActionRequest(req.to_owned()),
+    )
+    .await?;
+
+    let payout_create = payout_data.payout_create.to_owned();
+    let status = payout_create.status.foreign_into();
+
+    // Verify if fulfillment can be triggered
+    if helpers::is_payout_terminal_state(status) {
+        return Err(report!(errors::ApiErrorResponse::InvalidRequestData {
+            message: format!(
+                "{} is in terminal state [status: {}]",
+                payout_create.payout_id.to_owned(),
+                status
+            ),
+        }));
+    }
+
+    // TODO: Remove hardcoded connector
+    let connector_name = api_enums::Connector::Adyen;
+
+    // Form connector data
+    let connector_data: api::ConnectorData = api::ConnectorData::get_connector_by_name(
+        &state.conf.connectors,
+        &connector_name.to_string(),
+        api::GetToken::Connector,
+    )
+    .change_context(errors::ApiErrorResponse::InternalServerError)
+    .attach_printable("Failed to get the connector data")?;
+
+    // Trigger fulfillment
+    let pmd = payouts::PayoutMethodData::default(); // TODO: Fetch from locker
```

Comment:
add an issue for the locker handling part and link it here

Reply:
This is already fixed in a separate micro-PR